src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java

```diff
@@ -786,9 +789,68 @@ private static Operand translateCurrentTimestamp(final Instant currentTime) {
         @SneakyThrows
         private static Operand translateDateAdd(final RexCall call, final List<Operand> strings) {
             verifySupportedDateAddType(call.getOperands().get(1));
+
+            // Is date addition between literals (including CURRENT_DATE)?
+            final boolean isLiteralCandidate = isDateLiteralCandidate(call, strings);
+            if (isLiteralCandidate) {
+                // Perform in-memory calculation before sending to server.
+                return getDateAddLiteralOperand(strings);
+            }
+            // Otherwise, perform addition on server.
             return new Operand("{ \"$add\":" + "[" + Util.commaList(strings) + "]}");
         }
 
+        private static boolean isDateLiteralCandidate(final RexCall call, final List<Operand> strings) {
+            final boolean allLiterals = call.getOperands().stream()
+                    .allMatch(op -> {
+                        final SqlKind opKind = op.getKind();
+                        final String opName = op.toString();
+                        return opKind == SqlKind.LITERAL
```

Comment:
can we add unit tests operand is date or timestamp literal?

Reply:
OK.